edgelet/hsm-sys/azure-iot-hsm-c/src/certificate_info.c

```diff
@@ -602,10 +602,13 @@ void certificate_info_destroy(CERT_INFO_HANDLE handle)
     if (cert_info != NULL)
     {
         free(cert_info->first_certificate);
+        cert_info->first_certificate = NULL;
         free(cert_info->certificate_pem);
+        cert_info->certificate_pem = NULL;
         if (cert_info->private_key != NULL)
         {
             free(cert_info->private_key);
+            cert_info->private_key = NULL;
```

Comment:
Curious, aren't these `NULL` assignments redundant given that `cert_info` itself is going to be freed on line 613?

Reply:
There is a chance that a client could use a freed handle which might have these stale entries. By zeroing out, we eliminate some aspect of "undefined behavior" and the failure will be a consistent NULL ptr deref.